libraries/abstractions/transport/include/transport_interface.h

```diff
@@ -0,0 +1,203 @@
+/*
```

Comment:
There are offline concerns about this duplication. 
I believe depending on the demo, either coreMQTT or coreHTTP's copy of this header will be pull used, so this should be deleted.